source/js/main.js

```diff
@@ -49,19 +54,69 @@ function closeMenu() {
   tabIndexer();
 }
 
+// Google Analytics
+function initializeGA(trackingId) {
+  ga("create", trackingId, "auto");
+
+  // Ensure we don't pass the email query param to Google Analytics
+  var loc = window.location,
+    protocol = loc.protocol,
+    hostname = loc.hostname,
+    pathname = loc.pathname,
+    filteredQueryParams = loc.search
+      .substring(1)
+      .split("&")
+      .filter(param => !param.startsWith("email"))
+      .join("&");
+
+  ga(
+    "set",
+    "location",
+    `${protocol}//${hostname}${pathname}?${filteredQueryParams}`
+  );
+  ga("send", "pageview");
+  ga("require", "ecommerce");
+
+  // Check for any events sent by the view, and fire them.
+  var gaEventsNode = document.getElementById("ga-events");
+  if (gaEventsNode) {
+    var events = JSON.parse(gaEventsNode.textContent);
+    events.forEach(eventArray => {
+      ga(...eventArray);
+    });
+  }
+
+  // Click events
+  for (const a of document.querySelectorAll(".js-ga-track-click")) {
+    a.addEventListener("click", e => {
+      ga("send", "event", {
+        eventCategory: a.getAttribute("data-ga-category"),
+        eventAction: a.getAttribute("data-ga-action"),
+        eventLabel: a.getAttribute("data-ga-label"),
+        transport: "beacon"
+      });
+    });
+  }
+}
+
 document.addEventListener("DOMContentLoaded", function() {
-  // Initialize Sentry error reporting
+  // Initialize Sentry error reporting and analytics
 
   fetchEnv(envData => {
-    if (!envData.SENTRY_DSN) {
-      return;
+    const {
+      GA_TRACKING_ID: trackingId,
+      SENTRY_DSN: dsn,
+      RELEASE_VERSION: release,
+      SENTRY_ENVIRONMENT: environment
+    } = envData;
+
+    if (allowGA && typeof ga === "function" && trackingId) {
+      initializeGA(trackingId);
```

Comment:
Can we guarantee that the GA scripts will be loaded and ready to initialize before this section of code? Seems there's a bit of a race condition at first glance?

Reply:
if it doesn't, this code won't run -- hence the `typeof ga === "function"` check

Reply:
Right, but what if it should run, but doesn't, because the ga script tag injected below hasn't loaded yet? 

In that case, we're losing out on analytics we would have otherwise been able to use.

Reply:
My concern is that `DOMContentLoaded` is tied to the loading of the inital HTML document, so I don't think it's going to wait for that GA script tag.

Reply:
any suggestions on how to fix that?

Reply:
We can rearrange things a bit and hook  the initialization code into the scripts onload event? We might need to poll for `ga` though..

Reply:
yeah so onload doesn't guarantee that the script has run, just that it has loaded. Are you sure you want to introduce `setInterval` or `setTimeout` into this?